app/services/activitypub/verify_quote_service.rb

```
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+class ActivityPub::VerifyQuoteService < BaseService
+  include JsonLdHelper
+
+  # Optionally fetch quoted post, and verify the quote is authorized
+  def call(quote, fetchable_quoted_uri: nil, prefetched_body: nil, request_id: nil)
+    @request_id = request_id
+    @quote = quote
+    @fetching_error = nil
+
+    fetch_quoted_post_if_needed!(fetchable_quoted_uri)
+    return if fast_track_approval! || quote.approval_uri.blank?
+
+    @json = fetch_approval_object(quote.approval_uri, prefetched_body:)
+    if @json.nil?
+      # It is possible the proof isn't dereferenceable yet, so only reject if already accepted
+      quote.reject! unless @quote.pending?
+      return
```

This might need to raise an exception so we can reschedule the check.